src/xsdba/units.py

```diff
@@ -298,7 +298,25 @@ def pint2cfattrs(value: units.Quantity | units.Unit, is_difference=None) -> dict
     return attrs
 
 
-# XC simplified
+# Private function so it can be patched
+def _convert_units_to(  # noqa: C901
+    source: Quantified,
+    target: Quantified | units.Unit,
+) -> xr.DataArray | float:
+    target_unit = units2str(target)
+    source_unit = units2str(source)
+    if target_unit == source_unit:
+        return source if not isinstance(source, str) else float(str2pint(source).m)
+    else:  # Convert units
+        if isinstance(source, xr.DataArray):
+            out = source.copy(data=units.convert(source.data, source_unit, target_unit))
+            out = out.assign_attrs(units=target_unit)
+        else:  # scalar
+            # explicit float cast because cf-xarray registry outputting 0-dim arrays
+            out = float(str2pint(source).to(target_unit).m)
```

Comment:
Does this imply that `cf-xarray` behaviour should be patched?

Reply:
I should compare in more details with xclim.indices, there's nothing similar on the surface, so maybe something has already been patched but I didn't update xsdba